roles/openshift_health_checker/openshift_checks/disk_availability.py

```diff
@@ -81,6 +90,16 @@ def run(self, tmp, task_vars):
             config_bytes = max(config.get(name, 0) for name in group_names) * 10**9
             recommended_bytes = config_bytes or recommended_bytes
 
+            # if an "upgrade" context is set, update the minimum disk requirement
+            # as this signifies an in-place upgrade - the node might have the
+            # required total disk space, but some of that space may already be
+            # in use by the existing OpenShift deployment.
+            context = get_var(task_vars, "r_openshift_health_checker_playbook_context", default="")
+            if context == "upgrade":
+                recommended_upgrade_paths = self.recommended_disk_upgrade_bytes.get(path, {})
+                if recommended_upgrade_paths:
+                    recommended_bytes = max(recommended_upgrade_paths.get(name, 0) for name in group_names)
```

Comment:
I think this is still wrong because the user configuration should have precedence over everything else?
If you agree, please add a test case for that, see the fail, then fix the implementation and see the test pass. Commit and push :-)